### PR TITLE
[TU-13733] Add SonarCloud configuration

### PR DESCRIPTION
**Pull request by vrybas**
JIRA: https://typeform.atlassian.net/browse/TU-13733

Coverage reported [in SonarCloud](https://sonarcloud.io/summary/new_code?id=Typeform_embed&pullRequest=659), linked in [a comment below](https://github.com/Typeform/embed/pull/659#issuecomment-2288041027):
Screenshot 2024-08-14 at 09 32 59

### CODE REVIEW

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `56e9fd44db56db7c4afe05338704700697a18d9f` [here](https://embed.typeform.com/56e9fd44db56db7c4afe05338704700697a18d9f/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `8c80f26939f4f3bcecb5b4105ff6bebb2e87ac8e` [here](https://embed.typeform.com/8c80f26939f4f3bcecb5b4105ff6bebb2e87ac8e/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `f131875f1316d5b8dfe88b2a0139225a544b6e0a` [here](https://embed.typeform.com/f131875f1316d5b8dfe88b2a0139225a544b6e0a/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `1133440ca94cfb25a01da7b7b6f0a23096b8b2d1` [here](https://embed.typeform.com/1133440ca94cfb25a01da7b7b6f0a23096b8b2d1/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `b3308c3991af2089dfc91e84918676e83f628d3e` [here](https://embed.typeform.com/b3308c3991af2089dfc91e84918676e83f628d3e/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `6fdc5e1d75780876f97aab5decf4b66cbebbb405` [here](https://embed.typeform.com/6fdc5e1d75780876f97aab5decf4b66cbebbb405/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `b80f44d4142a9e09ed01a14258990cbe7cc69e02` [here](https://embed.typeform.com/b80f44d4142a9e09ed01a14258990cbe7cc69e02/embed.js).

File: packages/embed/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "lint": "eslint src --ext .js,.ts,.jsx,.tsx --max-warnings=0 && yarn prettier-check",
     "prettier-check": "prettier --check . --ignore-path .eslintignore",
     "prettier": "prettier --write . --ignore-path .eslintignore",
-    "test": "jest",
+    "test": "jest --coverage",
```
[the remark this answers is not in the document]
Comment (vrybas) [reply packages/embed/package.json:43]: well, there's a bit of a problem with this

`pull-request.yml` calls
```
42:       - run: yarn test
```

and it is translated to `yarn lerna run test` somewhere down the line

image

I tried to pass the `--coverage` flag to the initial `yarn test`, but `lerna` refuses to take it, and, in turn, to pass over to the other packages. 

I read somewhere you need to update the individual packages' commands.

So I decided to edit the command itself in `package.json` 

Hopefully FEs can help to refactor this at some point, but it seems to work 😄

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `07ac7eae5d7e4d0f53e1dc00d378bfa58634e198` [here](https://embed.typeform.com/07ac7eae5d7e4d0f53e1dc00d378bfa58634e198/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `6fc4ae46ab24a8dfccc386807dad981a2c3e318f` [here](https://embed.typeform.com/6fc4ae46ab24a8dfccc386807dad981a2c3e318f/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `dcd203a173a48d1a91d13e8a32a92178ed663d03` [here](https://embed.typeform.com/dcd203a173a48d1a91d13e8a32a92178ed663d03/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `dcd203a173a48d1a91d13e8a32a92178ed663d03` [here](https://embed.typeform.com/dcd203a173a48d1a91d13e8a32a92178ed663d03/embed.js).

File: packages/embed/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (mathio) [reply packages/embed/package.json:43]: @gierlas I like the idea. I will also rename the `coverage` script to `test:coverage` to be more explicit.

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `0c75885d18336592e95737e81c8efb8bb07de15d` [here](https://embed.typeform.com/0c75885d18336592e95737e81c8efb8bb07de15d/embed.js).

**Review by muloem — APPROVED** [review]

**Review by gierlas — APPROVED** [review]

**Pull request merged**